packages/twenty-front/src/modules/object-record/record-group/hooks/useCurrentRecordGroupId.ts

```diff
@@ -1,20 +1,37 @@
+/* eslint-disable no-redeclare */
+/* eslint-disable prefer-arrow/prefer-arrow-functions */
 import { RecordGroupContext } from '@/object-record/record-group/states/context/RecordGroupContext';
 import { useContext } from 'react';
+import { isDefined } from '~/utils/isDefined';
 
-export const useCurrentRecordGroupId = () => {
+type UseCurrentRecordGroupIdOptions = {
+  allowUndefined?: boolean;
+};
+
+export function useCurrentRecordGroupId(): string;
```

Comment:
looks over complex to me why do we need overload here?

Reply:
Agree, function overloading should be used only for very specific utils / internal API because it's complex to understand and maintain for the average developer and we can pretty much find a simpler API in the vast majority of cases.